codepy/bpl.py

```diff
@@ -166,7 +168,8 @@ def compile(self, toolchain, **kwargs):
         """
 
         from codepy.libraries import add_boost_python
-        toolchain = toolchain.copy()
+
+        toolchain = replace(toolchain)
```

Comment:
Also not sure if this replace does what it should. All libraries, includes, etc, in `toolchain` are lists and this just performs a shallow copy..